src/vdom/core.cljs

```diff
@@ -26,12 +26,17 @@
 
 (declare svg-tree)
 
+(defn virtual-node? [arg]
+  ;; use javascript notion of true/false,
+  ;; otherwise, this will return true for "" and cause issues
+  (js/Boolean (js/VDOM.isVirtualNode arg)))
+
 (defn html-tree [arg]
   (cond
     (nil? arg)
```

Comment:
Maybe just make this check and the one below `(or (nil? arg) (= arg ""))`?

The above `virtual-node?` function is confusing to me, since I'd expect `js/VDOM.isVirtualNode` to always return true or false. Is that not the case?

Reply:
https://github.com/Matt-Esch/virtual-dom/blob/947ecf92b67d25bb693a0f625fa8e90c099887d5/vnode/is-vnode.js#L6

> The above virtual-node? function is confusing to me, since I'd expect js/VDOM.isVirtualNode to always return true or false. Is that not the case?



That is not the case.`isVirtualNode` will return any falsey value that it receives (eg. "") . This will cause issues for any value that is false in js, but true in cljs.

From https://developer.mozilla.org/en-US/docs/Glossary/Falsy
That includes:
0 | The number zero
-0 | The number negative zero
0n | BigInt, when used as a boolean, follows the same rule as a Number. 0n is falsy.
"" | Empty string value
NaN | NaN - not a number

I believe the best and most direct fix is to explicitly coerce to a boolean(true/false) which is what `js/Boolean` does.